dipy/align/tests/test_imaffine.py

```diff
@@ -172,7 +172,7 @@ def test_align_origins_3d():
 
 def test_affreg_all_transforms():
     # Test affine registration using all transforms with typical settings
-    for ttype in factors.keys():
+    for ttype in sorted(factors):
```

Comment:
Comment to explain why the factor keys must be sorted (in order to preserve relationship of random numbers to dict key / values)?  Ditto for other instances.